### PR TITLE
ESQL: Emit `null` for missing `_source` fields

### PR DESCRIPTION
**Pull request by nik9000**
When ESQL loads from `_source` it was accidentally emitting empty arrays instead of null values for missing fields. These come out as `[]` for missing fields but should be `null`.

### CODE REVIEW

**Comment by nik9000** [comment]
Labeled non-issue because it's fixing an issue in unreleased code.

File: x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/data/SingletonOrdinalsBuilderTests.java [hunk 1]
```diff
@@ -122,6 +122,37 @@ protected DriverContext breakingDriverContext() { // TODO move this to driverCon
         return new DriverContext(bigArrays, factory);
     }
 
+    public void testAllNull() throws IOException {
+        BlockFactory factory = breakingDriverContext().blockFactory();
```
Comment (dnhatn) [inline x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/data/SingletonOrdinalsBuilderTests.java:126]: I am not sure I understand the purpose of this test?

**Review by dnhatn — APPROVED** [review]
LGTM.

File: x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/data/SingletonOrdinalsBuilderTests.java (hunk 1 above)
Comment (nik9000) [reply x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/data/SingletonOrdinalsBuilderTests.java:126]: Ah, sorry. It's only vaguely related. I was investigating this issue and thought that the cause was ordinals not stored fields. But it turns out that ordinals were fine. Still, I wanted to make sure this was all true. Just that when you use the ordinals builder if you only give it nulls it'll build something that properly declares things as null.

**Pull request merged**